src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/RegisterChecks/RoepaoTests.cs

```diff
@@ -0,0 +1,142 @@
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AdminService.Web.Tests.Controllers.Gateway.RegisterChecks
+{
+    [TestFixture]
+    public class RoepaoTests
+    {
+        private RoatpGatewayRegisterChecksController _controller;
+        private Mock<IRoatpApplicationApiClient> _applyApiClient;
+        private Mock<IHttpContextAccessor> _contextAccessor;
+        private Mock<IRoatpGatewayPageViewModelValidator> _gatewayValidator;
+        private Mock<IGatewayRegisterChecksOrchestrator> _orchestrator;
+        private Mock<ILogger<RoatpGatewayRegisterChecksController>> _logger;
+
+        private string username = "john smith";
+        private string givenName = "john";
+        private string surname = "smith";
+
+        [SetUp]
+        public void Setup()
+        {
+            _applyApiClient = new Mock<IRoatpApplicationApiClient>();
+            _contextAccessor = new Mock<IHttpContextAccessor>();
+            _gatewayValidator = new Mock<IRoatpGatewayPageViewModelValidator>();
+            _logger = new Mock<ILogger<RoatpGatewayRegisterChecksController>>();
+            _orchestrator = new Mock<IGatewayRegisterChecksOrchestrator>();
+
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]
+             {
+                new Claim(ClaimTypes.NameIdentifier, "1"),
+                new Claim("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn", username),
+                new Claim(ClaimTypes.GivenName, givenName),
+                new Claim(ClaimTypes.Surname, surname)
+             }));
+
+            _contextAccessor.Setup(_ => _.HttpContext).Returns(new DefaultHttpContext { User = user });
+
+            _gatewayValidator.Setup(v => v.Validate(It.IsAny<RoepaoPageViewModel>()))
+                .ReturnsAsync(new ValidationResponse
+                {
+                    Errors = new List<ValidationErrorDetail>()
+                });
+
+            _controller = new RoatpGatewayRegisterChecksController(_applyApiClient.Object, _contextAccessor.Object, _gatewayValidator.Object, _orchestrator.Object, _logger.Object);
+        }
+
+        [Test]
+        public async Task check_roepao_request_is_sent()
+        {
+            var applicationId = Guid.NewGuid();
+
+            _orchestrator.Setup(x => x.GetRoepaoViewModel(new GetRoepaoRequest(applicationId, username)))
+                .ReturnsAsync(new RoepaoPageViewModel())
+                .Verifiable("view model not returned");
+
+            var _result = await _controller.GetGatewayRoepaoPage(applicationId, GatewayPageIds.Roepao);
+            _orchestrator.Verify(x => x.GetRoepaoViewModel(It.Is<GetRoepaoRequest>(y => y.ApplicationId == applicationId && y.UserName == username)), Times.Once());
+        }
+
+        [Test]
+        public async Task roepao_viewmodel_is_returned()
```

Comment:
Can you rename to avoid technical implementation details (viewmodel)?

Reply:
The other tests follow the same pattern 🤷‍♂